app/src/main/java/org/fossasia/pslab/communication/sensors/Sx1276.java

```diff
@@ -0,0 +1,454 @@
+    // Registers
+    private int REG_FIFO = 0x00;
+    private int REG_OP_MODE = 0x01;
+    private int REG_FRF_MSB = 0x06;
+    private int REG_FRF_MID = 0x07;
+    private int REG_FRF_LSB = 0x08;
+    private int REG_PA_CONFIG = 0x09;
+    private int REG_LNA = 0x0c;
+    private int REG_FIFO_ADDR_PTR = 0x0d;
+    private int REG_FIFO_TX_BASE_ADDR = 0x0e;
+    private int REG_FIFO_RX_BASE_ADDR = 0x0f;
+    private int REG_FIFO_RX_CURRENT_ADDR = 0x10;
+    private int REG_IRQ_FLAGS = 0x12;
+    private int REG_RX_NB_BYTES = 0x13;
+    private int REG_PKT_RSSI_VALUE = 0x1a;
+    private int REG_PKT_SNR_VALUE = 0x1b;
+    private int REG_MODEM_CONFIG_1 = 0x1d;
+    private int REG_MODEM_CONFIG_2 = 0x1e;
+    private int REG_PREAMBLE_MSB = 0x20;
+    private int REG_PREAMBLE_LSB = 0x21;
+    private int REG_PAYLOAD_LENGTH = 0x22;
+    private int REG_MODEM_CONFIG_3 = 0x26;
+    private int REG_RSSI_WIDEBAND = 0x2c;
+    private int REG_DETECTION_OPTIMIZE = 0x31;
+    private int REG_DETECTION_THRESHOLD = 0x37;
+    private int REG_SYNC_WORD = 0x39;
+    private int REG_DIO_MAPPING_1 = 0x40;
+    private int REG_VERSION = 0x42;
+    private int REG_PA_DAC = 0x4D;
+    // Modes
+    private int MODE_LONG_RANGE_MODE = 0x80;
+    private int MODE_SLEEP = 0x00;
+    private int MODE_STDBY = 0x01;
+    private int ODE_STDBY = 0x01;
+    private int MODE_TX = 0x03;
+    private int MODE_RX_CONTINUOUS = 0x05;
+    private int MODE_RX_SINGLE = 0x06;
+    // PA config
+    private int PA_BOOST = 0x80;
+    // IRQ masks
+    private int IRQ_TX_DONE_MASK = 0x08;
+    private int IRQ_PAYLOAD_CRC_ERROR_MASK = 0x20;
+    private int IRQ_RX_DONE_MASK = 0x40;
+
+    private int MAX_PKT_LENGTH = 255;
+    private int PA_OUTPUT_RFO_PIN = 0;
+    private int PA_OUTPUT_PA_BOOST_PIN = 1;
+    private int onReceive = 0;
+    private int frequency = 10;
+    private int packetIndex = 0;
+    private int packetLength = 0;
+    private int version = 0;
+    private int implicitHeaderMode = 0;
+
+    private SPI spi;
+    private ArrayList<Byte> bytes = new ArrayList<>();
+
+    public Sx1276(SPI spi, int frequency, int power, boolean boost, double bw, int sf, int cf) throws IOException {
+        initiateSX1276(spi, frequency);
+        // Output Power 17dbm
+        setTxPower(power, boost ? PA_OUTPUT_PA_BOOST_PIN : PA_OUTPUT_RFO_PIN);
+        idle();
+        // Set bandwidth
+        setSignalBandwidth(bw);
+        setSpreadingFactor(sf);
+        setCodingRate4(cf);
+    }
+
+    public Sx1276(SPI spi, int frequency) throws IOException {
+
+        initiateSX1276(spi, frequency);
+        // Output Power 17dbm
+        setTxPower(17, PA_OUTPUT_RFO_PIN);
+        idle();
+        // Set bandwidth
+        setSignalBandwidth(125e3);
+        setSpreadingFactor(12);
+        setCodingRate4(5);
+    }
+
+    private void initiateSX1276(SPI spi, int frequency) throws IOException {
+
+        this.spi = spi;
+        this.spi.setParameters(2, 6, 1, 0, 1);
```

Comment:
We should re-implement the set_parameters method.
It should accept a frequency argument instead of two separate prescalers and a developer trying to figure out their values given that the operating clock is 64MHz. the issue will go into both pslab-android, as well as pslab-python
optional : CKE,CKP can be replaced with CPHA, CPOL where CPHA= !CKE  . The latter is a more commonly used notation. just a thought.